### ISSUE TITLE
incorrect nemolite2d metadata

### ISSUE THREAD

**Issue by rupertford** [stfc/PSycloneBench#5]
The boundary condition routines bc_solid_u and bc_solid_v have their modified fields specified as being "write" in their PSyclone metadata. However, this should be "readwrite".

The reason these routines are declared as "write" is that the data that is modified is only written to. However, only some of the data is written to (which is determined by a mask) and the data that is not modified needs to remain intact from entry to exit of the routine. If the data is declared as "write" then this might not be the case, therefore we need the metadata to be "readwrite"

**Comment by rupertford** [comment]
The two cases where the metadata has changed and affects PSyclone code generation are nemo/psykal and nemo/psykal_omp.

I've run the nemo/psykal version with and without this change and the code produced by PSyclone is the same in both cases.

I looked at comparing PSyclone generated code nemo/psykal_omp version but the Makefile does not seem to be set up to use it. I got  ... `Code generation not yet working for version 0.2 of the GOcean API!` as output from the Makefile.
